core/modules/util/ThreadPool.cc

```diff
@@ -300,23 +300,26 @@ void ThreadPool::waitForResize(int millisecs) {
 void ThreadPool::_incrPoolThreadCount() {
     std::lock_guard<std::mutex> lockPool(_mxPool);
     ++_poolThreadCount;
-    LOGS(_log, LOG_LVL_WARN, "&&&WARN incr _poolThreadCount=" << _poolThreadCount);
+    LOGS(_log, LOG_LVL_DEBUG, "incr _poolThreadCount=" << _poolThreadCount);
 }
 
 void ThreadPool::_decrPoolThreadCount() {
     {
         std::lock_guard<std::mutex> lockPool(_mxPool);
         --_poolThreadCount;
     }
-    LOGS(_log, LOG_LVL_WARN, "&&&WARN decr _poolThreadCount=" << _poolThreadCount);
+    LOGS(_log, LOG_LVL_DEBUG, "decr _poolThreadCount=" << _poolThreadCount);
     _cvPool.notify_all(); // There should not be many threads waiting.
 }
 
 void ThreadPool::_waitIfAtMaxThreadPoolCount() {
     std::unique_lock<std::mutex> lockPool(_mxPool);
-    LOGS(_log, LOG_LVL_WARN, "&&&WARN wait A _poolThreadCount=" << _poolThreadCount);
+    auto logLvl = LOG_LVL_DEBUG;
+    if (_poolThreadCount >= _maxThreadCount - 10) {
```

Comment:
Why this fixed number `10` here? Does it have any specific meaning?
There is also a chance of getting into an integer overflow since is defined as:
```
unsigned int _maxThreadCount;
```
This would happed if, for example:
```
_maxThreadCount = 9;
```

Reply:
This just has to do with leaving a log message when approaching the limit. If max thread count is 9, all the messages will be warnings instead of debug. I'll just remove it.